app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [ ] [ optional] @yayaismaail Great work adding  a method that updates the posts counter for a user, however it would be nice if you used the[Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_user_posts_counter` method every time a new post has been saved to database, then make the `update_user_posts_counter` method private, you can also use it in your comment model.
```suggestion
  after_save :update_user_posts_counter

  private

  def update_user_posts_counter
    author.increment!(:posts_counter)
  end
```
```